### ISSUE TITLE
Clonning to block device should allow partition to be bigger than the archive

### ISSUE THREAD

**Issue by Jonathas-Conceicao** [oll3/bita#15]
When testing bita for applying changes to a ext4 formatted I've came across the following error:

```
Error: Size of output device (1430 MiB (1499661312 bytes)) differ from size of archive target file (142 MiB (149934080 bytes))
```

<details><summary>full command output</summary>

```
root@uh-qemu-x86-64:~# bita clone --seed-output updatehub-image.ext4.bita /dev/vda3
Archive:
  Built with version: 0.7.1
  Archive size: 29 MiB (30866271 bytes)
  Header checksum: 1635e9a121ca79f2cc1fb5e1bf23da9c8a64a5d95e146ba6299c900200d74f0cede30d88f6589abd34d9d07aef580a3d1041b3e722fc3ebc69a12421e03d0511
  Chunking algorithm: RollSum
  Rolling hash window size: 64 bytes
  Chunk minimum size: 16 KiB (16384 bytes)
  Chunk maximum size: 16 MiB (16777216 bytes)
  Chunk average target size: 64 KiB (65536 bytes) (mask: 0b111111111111111)
Source:
  Source checksum: cf895d5a02488c426a6a60dfda220f201a19bc1ba0c7eb4a5c8113319b21e81b2f0bc45923b5d3d21ebc288c03ed134fea5b76a2e42f9c5dc995177bef6e68b0
  Chunks in source: 1463 (unique: 1455)
  Average chunk size: 100 KiB (102909 bytes)
  Source size: 142 MiB (149934080 bytes)

Cloning archive updatehub-image.ext4.bita to /dev/vda3...
Error: Size of output device (1430 MiB (1499661312 bytes)) differ from size of archive target file (142 MiB (149934080 bytes))
```

</details>

Which was caused by the following piece of code:
https://github.com/oll3/bita/blob/e6d3a9efb4d133a4d4273b1b1608ee8c5a320d1d/src/clone_cmd.rs#L136-L145

I believe that error should only be raised in case that the partition is smaller than the archive. A bigger partition should not be an issue.

Commenting out that block of the code in order to test the clone as a hole did work for applying the desired changes, but indexing the partition took more time than I would expect. I think the hole output device is being processed in order to perform the clone. If that's the case, wouldn't it be possible (for cases such as this) to index only the chunks up to the archive total source size?

**Comment by oll3** [comment]
> I missed the case where a previous image was larger and we could, possibly, reuse some blocks after the end of the new file range. One thing I am wondering is: Do you foresee a way to "know" the size of the previous image?

I have been thinking about building some kind of cache functionality. Where the chunk index of a written output would be stored on clone. That way the seed scan could be skipped completely next clone when/if a cache for that seed file is available. Still one would verify chunk checksums so that the cache and the seed file hasn't diverged. Then of course one need to decide what to do on checksum mismatch, fall back to a regular scan or be done or else.
Other than that no. And have not started this work yet and not sure when I will have time to do neither.

**Comment by oll3** [comment]
Closing since fixed from release 0.9.0.
